java/server/src/org/openqa/selenium/grid/distributor/local/GridModel.java

```diff
@@ -63,10 +60,6 @@ public GridModel(EventBus events) {
     this.events = Require.nonNull("Event bus", events);
 
     this.events.addListener(NodeDrainStarted.listener(nodeId -> setAvailability(nodeId, DRAINING)));
-    this.events.addListener(NodeDrainComplete.listener(this::remove));
-    this.events.addListener(NodeRemovedEvent.listener(this::remove));
```

Comment:
I only see this event being listened in the `GridModel`, did I miss something?

Reply:
https://github.com/SeleniumHQ/selenium/blob/5004ca80649e6bae92b1f2c210b3feb21643150d/java/server/src/org/openqa/selenium/grid/distributor/local/LocalDistributor.java#L283 
The method here explicitly calls the Grid Model remove() method then fires the NodeRemovedEvent event which again calls the same Grid Model method. Hence, I thought it is redundant since the Grid Model is part of the Local Distributor.

I see that the NodeRemovedEvent is only used in tests and the Grid Model. Now I am curious to understand the use of NodeRemovedEvent. I might be missing something here.

Reply:
I see what you mean, I will double check.